docs/TRANSLATIONS.md

```diff
@@ -0,0 +1,152 @@
+
+Thank you for your interest in contributing translations to the RetroAchievements website! By helping translate our platform, you make it more accessible to a wider audience. This guide will walk you through the steps to add or update translations in the RAWeb repository.
+
+## Table of Contents
+
+- [Prerequisites](#prerequisites)
+- [Getting Started](#getting-started)
+- [Translation Guidelines](#translation-guidelines)
+  - [File Structure](#file-structure)
+  - [Maintaining Keys](#maintaining-keys)
+  - [Handling Placeholders](#handling-placeholders)
+- [Submitting Your Changes](#submitting-your-changes)
+- [Review Process](#review-process)
+- [Additional Resources](#additional-resources)
+
+## Prerequisites
+
+Before you begin, ensure you have the following:
+
+- **GitHub Account:** You'll need to fork the RAWeb repository and submit pull requests.
+- **Basic Git Knowledge:** Familiarity with forking repositories, creating branches, committing changes, and opening pull requests.
+- **JSON Editing Skills:** Ability to read and edit JSON files without introducing syntax errors.
+- **Language Proficiency:** Proficiency in the target language to ensure accurate and meaningful translations.
+
+## Getting Started
+
+1. **Fork the RAWeb Repository**
+
+- Navigate to the [RAWeb GitHub repository](https://github.com/RetroAchievements/RAWeb).
+- Click the **Fork** button at the top-right corner to create a personal copy of the repository associated with your GitHub account.
+
+2. **Clone Your Fork:**
+
+```shell
+git clone https://github.com/your-username/RAWeb.git
+cd RAWeb
+```
+
+3. **Create a New Branch:**
+
+It's a best practice to create a separate branch for your translation work.
+
+```shell
+git checkout -b translate/pt_BR
+```
+
+## Translation Guidelines
+
+### File Structure
+
+All translations are stored in the _lang_ directory as JSON files. Each language has its own file, named using the [IETF language tag](https://en.wikipedia.org/wiki/IETF_language_tag) format, such as _en_US.json_ for American English or _pt_BR.json_ for Brazilian Portuguese.
+
+**Example Path:**
+
+```
+RAWeb/
+└── lang/
+    ├── en_US.json
+    └── pt_BR.json
+```
+
+### Maintaining Keys
+
+- **Do Not Alter Keys:** Each entry in the JSON file consists of a key-value pair. **Only translate the values.** The keys are used by the application to reference the correct text and must remain unchanged.
+
+```json
+{
+  ":count posts in the last 24 hours": ":count posts nas últimas 24 horas"
+}
+```
+
+- **Consistent Formatting:** Ensure that the structure and formatting (such as indentation) of the JSON file remaining consistent with the original.
+
+### Handling Placeholders
+
+Some translation strings contain placeholders (eg: `:count`, `:total`, `:gameTitle`). These placeholders are dynamically replaced by the application at runtime.
+
+- **Keep Placeholders Intact:** Do not translate or modify placeholders. Ensure they remain exactly as they are in the original key.
```

Comment:
Is there some way to explicitly delimit the placeholder? Some languages have counter marks that follow numbers. Here's a contrived example using English:

`:{count}x new posts` => `5x new posts`

Reply:
For better or worse, we're tightly coupled to how Laravel itself manages i18n strings, and it does not support delimiters.

However, unexpectedly, it appears it's smart enough to handle these cases. Taking your example:
```json
":countx new posts": ":countx new posts"
```
```tsx
<p>{t(':countx new posts', { count: 5 })}</p>
```
Correctly renders:
![Screenshot 2024-10-23 at 9 23 01 AM](https://github.com/user-attachments/assets/7338b449-31b9-4bc9-878e-bff04f08c47b)

The challenge is it may not really be clear to a translator that "x" isn't a placeholder.

It may also be possible to handle this with the component introduced in https://github.com/RetroAchievements/RAWeb/pull/2774 with something like:
```json
"<0>:count</0>x new posts": "<0>:count</0>x new posts"
```